src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -120,6 +120,47 @@ private ImmutableArray<ISymbol> GetSymbolsForCurrentContext()
             return ImmutableArray<ISymbol>.Empty;
         }
 
+        private static bool IsInsideRightHandOfPatternExpressionSyntax(SyntaxNode? node)
+        {
+            var previousNode = node;
+            node = node?.Parent;
+            while (node is not null)
+            {
+                var kind = node.Kind();
+                if (SyntaxFacts.IsAccessorDeclaration(kind) ||
```

Comment:
accessordeclaration??  whats' the case where this matters?

it looks like you're just bailing out.

but i would expect to see the set of things you do *not* bail on that allows you to walk upwards.  the existing code ahs the same problem as before, it just doesn't walk all the way up to teh root.  It's not the walking to the root that's the problem.  It's taht it walks up arbitrary nodes in the middle whose cases we haven't explicitly decided we need to support.

Reply:
So the intended approach would be to explicitly support and permit specific nodes that may be present under an IsExpressionPatternSyntax subtree.

Reply:
and explicitly all that is minimally necessary to support this case.  My presumption is, for example, once the compiler detects this is constant pattern it will parse this in a way where normal code falls out.

TBH, this seems pretty simple afaict.  You just need to check this spine afaict:

```
is X.Y.$$
```

In this case (and only this case afaict) you'd need to special case things.  If you had: `is X.Y.Z and A.` i would expect this was a normal member-access-expression not a qualified-name anymore.